code/messaging/src/main/java/com/adobe/marketing/mobile/aepcomposeui/components/LargeImageCard.kt

```diff
@@ -0,0 +1,151 @@
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.aepcomposeui.components
+
+import android.graphics.Bitmap
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.isSystemInDarkTheme
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.size
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.compose.ui.graphics.painter.BitmapPainter
+import androidx.compose.ui.unit.dp
+import com.adobe.marketing.mobile.aepcomposeui.AepUIConstants
+import com.adobe.marketing.mobile.aepcomposeui.LargeImageUI
+import com.adobe.marketing.mobile.aepcomposeui.UIAction
+import com.adobe.marketing.mobile.aepcomposeui.UIEvent
+import com.adobe.marketing.mobile.aepcomposeui.observers.AepUIEventObserver
+import com.adobe.marketing.mobile.aepcomposeui.style.LargeImageUIStyle
+import com.adobe.marketing.mobile.messaging.ContentCardImageManager
+
+/**
+ * Composable function that renders a large image card UI.
+ *
+ * @param ui The large image AEP UI to be rendered.
+ * @param style The style to be applied to the large image UI.
+ * @param observer An optional observer that listens to UI events.
+ */
+@Composable
+fun LargeImageCard(
+    ui: LargeImageUI,
+    style: LargeImageUIStyle,
+    observer: AepUIEventObserver?,
+) {
+    var isLoading by remember { mutableStateOf(true) }
+    var imageBitmap by remember { mutableStateOf<Bitmap?>(null) }
+    val imageUrl = if (isSystemInDarkTheme() && ui.getTemplate().image?.darkUrl != null)
+        ui.getTemplate().image?.darkUrl else ui.getTemplate().image?.url
+
+    LaunchedEffect(ui.getTemplate().id) {
+        observer?.onEvent(UIEvent.Display(ui))
+        if (imageUrl.isNullOrBlank()) {
+            isLoading = false
+        } else {
+            ContentCardImageManager.getContentCardImageBitmap(imageUrl) {
+                it.onSuccess { bitmap ->
+                    imageBitmap = bitmap
+                    isLoading = false
+                }
+                it.onFailure {
+                    // todo - confirm default image bitmap to be used here
+                    // imageBitmap = contentCardManager.getDefaultImageBitmap()
+                    isLoading = false
+                }
+            }
+        }
+    }
```

Comment:
Since we are now handling image loading for more than one card, can we use this opportunity to self contain the loading + displaying logic into an internal image view implementation? That way these UI's can just include it and not bother about the waiting + loading logic?